py3-setuptools-git-versioning.yaml

```diff
@@ -0,0 +1,114 @@
+package:
+  name: py3-setuptools-git-versioning
+  version: 2.0.0
+  epoch: 0
+  description: Use git repo data for building a version number according PEP-440
+  copyright:
+    - license: MIT
+  dependencies:
+    provider-priority: 0
+
+vars:
+  pypi-package: setuptools-git-versioning
+  import: setuptools_git_versioning
+
+data:
+  - name: py-versions
+    items:
+      3.10: "310"
+      3.11: "311"
+      3.12: "312"
+      3.13: "300"
+
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - py3-supported-cython
+      - py3-supported-pip
+      - py3-supported-python-dev
+      - py3-supported-toml
+      - wolfi-base
+
+subpackages:
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}
+    description: python${{range.key}} version of ${{vars.pypi-package}}
+    dependencies:
+      provider-priority: ${{range.value}}
+      provides:
+        - py3-${{vars.pypi-package}}
+      runtime:
+        - py${{range.key}}-packaging
+        - py${{range.key}}-setuptools
+        - py${{range.key}}-toml
+    pipeline:
+      - uses: py/pip-build-install
+        with:
+          python: python${{range.key}}
+      - name: "move usr/bin executables for -bin"
+        runs: |
+          mkdir -p ./cleanup/${{range.key}}/
+          mv ${{targets.contextdir}}/usr/bin ./cleanup/${{range.key}}/
+      - uses: strip
+    test:
+      pipeline:
+        - uses: python/import
+          with:
+            python: python${{range.key}}
+            import: ${{vars.import}}
+
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}-bin
+    description: Executable binaries for ${{vars.pypi-package}} installed for python${{range.key}}
+    dependencies:
+      runtime:
+        - py${{range.key}}-${{vars.pypi-package}}
+      provides:
+        - py3-${{vars.pypi-package}}-bin
+        - py3-${{vars.pypi-package}}
+      provider-priority: ${{range.value}}
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.contextdir}}/usr/
+          mv ./cleanup/${{range.key}}/bin ${{targets.contextdir}}/usr/
+    test:
+      pipeline:
+        - runs: setuptools-git-versioning --help
+
+  - name: py3-supported-${{vars.pypi-package}}
+    description: meta package providing ${{vars.pypi-package}} for supported python versions.
+    dependencies:
+      runtime:
+        - py3.10-${{vars.pypi-package}}
+        - py3.11-${{vars.pypi-package}}
+        - py3.12-${{vars.pypi-package}}
+        - py3.13-${{vars.pypi-package}}
+    test:
+      pipeline:
+        - uses: python/import
+          with:
+            python: python3.10
+            import: ${{vars.import}}
+
+pipeline:
+  - uses: git-checkout
+    with:
+      expected-commit: 1a2b9d0674b881e04b2a1238372b74020558274e
+      repository: https://github.com/dolfinus/setuptools-git-versioning
+      tag: v${{package.version}}
+
+  - name: Python Build
+    uses: py/pip-build-install
```

Comment:
```yaml
        with:
          python: python${{range.key}}
```

Reply:
Actually, do we just need to kill this section now that the builds are happening in the `subpackages:` ?

Reply:
actually, yes. good catch.

Reply:
and can you move the pipeline above the subpackages ? i really wich 'yam' would do all the things, but it does not.

Reply:
Could we add a jsonschema for the YAML, and and a `pre-commit` hook to validate it? That could enforce ordering, as well as using `yam`

Reply:
huh. i did not realize jsonschema could enforce ordering. I opened https://github.com/chainguard-dev/yam/issues/71 a while ago, I think people would generally be open to computer-enforced formatting (ie, 'go fmt').

Reply:
You're right - `jsonschema` doesn't enforce ordering (at least, not AFAIK). What I should've said is that I've written `pre-commit` hooks to enforce that YAML keys are sorted in the order that they are defined in a `jsonschema` file. I suppose the schema isn't strictly needed for the sorting, but schema validation is useful as well.